### DISCUSSION TITLE
Request for Assistance: Fortran Runtime Error in mHM Hydrologic Model

### DISCUSSION THREAD

**Discussion by ankur9217** [mhm-ufz/mHM#99] [Installation] [answered]
I am writing to seek assistance regarding an issue I encountered while running the mHM hydrologic model. The specific error message is as follows:

Fortran runtime error: Bad integer for item 2072 in list input
Error termination. Backtrace:
#0  0x7f77a5c0b8d4 in read_integer
        at ../../../libgfortran/io/list_read.c:1098
#1  0x7f77a5c0d678 in list_formatted_read_scalar
        at ../../../libgfortran/io/list_read.c:2170
#2  0x5606b1300121 in ???
#3  0x5606b132b281 in ???
#4  0x5606b127c057 in ???
#5  0x5606b127a253 in ???
#6  0x7f77a57afd8f in ???
#7  0x7f77a57afe3f in ???
#8  0x5606b127a2b0 in ???

To assist in troubleshooting, I have attached the input files necessary for running the model in a zip format. Please review the files and provide guidance on resolving this issue at your earliest convenience.
Thank you very much for your attention to this matter. I look forward to your prompt response.

Regards,
Ankur Kumar
[Uploading morph - Copy.zip…]()

![error_mhm](https://github.com/mhm-ufz/mHM/assets/174605396/5b07665a-4de2-4ae4-9bd5-5449437138b6)

**Comment by pallavhydro** [comment]
Hi Ankur 

The error message is pointed towards LAI_Class.asc file. The download link to your zip file is not working here. Could you please re-upload it more reliably. That would allow all participants here to help you.

Pallav

**Reply by ankur9217** [reply]
Dear Pallav,
Thank you for bringing this to my attention. I apologize for the inconvenience caused by the broken download link. Please find the updated link to the zip file containing the LAI_Class.asc file below.
I appreciate your willingness to assist, and I look forward to your insights on resolving the issue with the mHM model.
Best Regards,
Ankur Kumar
[LAI _ Class.zip](https://github.com/user-attachments/files/16143153/LAI._.Class.zip)

**Reply by pallavhydro** [reply]
Any tips for Ankur @rohinikumar82 @StephanThober ?

**Reply by ankur9217** [reply]
> Any tips for Ankur @rohinikumar82 @StephanThober ?

Dear Pallav, I am waiting for your reply. Further I have attempted to regrid the lai.nc file with 12 month mean lai value but I am getting error message as " Read data ...
  Reading data ...
    Reading dem and lcover ...
      Reading dem for domain: 1 ...
      Reading lcover for domain: 1 ...
    Reading data for domain: 1 ...
      Reading slope ...
      Reading aspect ...
      Reading soil ids ...
      Reading LAI ...
 Variable name: lai
 ERROR*** type of variable does not match argument type. subroutine Get_NcVar". Please find
[lai.zip](https://github.com/user-attachments/files/16244879/lai.zip)
 enclosed regrided lai.nc file. Thank you in advance

**Reply by StephanThober** [reply, marked as answer]
Dear @ankur9217,

I had a look into the LAI_class.asc file you provided. This file should contain class ids (i.e., integers) referring to the corresponding class in the class_definition.txt. In your file, however, there a floating point numbers contained (maybe mean LAI values), hence the error message. Please correct the file and save the correct class ids in the LAI_class.asc.

Best,
Stephan

p.s.: For reference, these are the floating point numbers

array([-9.9990e+03,  6.6710e-01,  7.4290e-01,  8.8580e-01,  9.9150e-01,
        1.0087e+00,  1.0143e+00,  1.0586e+00,  1.1043e+00,  1.1072e+00,
        1.1103e+00,  1.1529e+00,  1.1900e+00,  1.1943e+00,  1.1972e+00,
        1.2015e+00,  1.2215e+00,  1.2457e+00,  1.2515e+00,  1.2615e+00,
        1.2701e+00,  1.2715e+00,  1.2801e+00,  1.2844e+00,  1.2860e+00,
        1.3129e+00,  1.3215e+00,  1.3273e+00,  1.3315e+00,  1.3587e+00,
        1.3886e+00,  1.3901e+00,  1.4101e+00,  1.4201e+00,  1.4629e+00,
        1.4701e+00,  1.4715e+00,  1.4716e+00,  1.4744e+00,  1.5015e+00,
        1.5072e+00,  1.5501e+00,  1.5558e+00,  1.5630e+00,  1.5886e+00,
        1.5887e+00,  1.6343e+00,  1.6458e+00,  1.7202e+00,  1.7516e+00,
        1.8517e+00,  1.8629e+00,  1.9273e+00,  2.0701e+00,  2.1062e+00,
        2.1472e+00,  2.2073e+00,  2.2488e+00,  2.2677e+00,  2.2787e+00,
        2.2875e+00,  2.4344e+00,  2.6088e+00,  2.6388e+00,  2.7373e+00,
        2.7429e+00,  2.7673e+00,  2.7773e+00,  2.7873e+00,  2.7916e+00,
        2.9788e+00,  3.0701e+00])
